src/main/java/de/tum/in/www1/exerciseapp/service/ExerciseService.java

```diff
@@ -138,11 +138,6 @@ public void reset(Exercise exercise) {
     @Transactional
     public void delete(Long id) {
         log.debug("Request to delete Exercise : {}", id);
-        Exercise exercise = exerciseRepository.findOne(id);
-
-        if (Optional.ofNullable(exercise).isPresent()) {
-            reset(exercise);
-        }
         exerciseRepository.delete(id);
```

Comment:
Why did you remove this code? It is essential, and the delete will otherwise not work. Please add it back!

Reply:
I moved this code to the ExerciseResource, please check line 227.  I anyway need to retrieve Exercise from ExerciseRepository in order to check authorization and, in order not to call ExerciseRepository.findOne(id) twice, I've moved this code to the resource.